base-console-app-cert/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Graph;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+
+namespace ConsoleGraphTest
+{
+    // This class encapsulates the details of getting a token from MSAL and exposes it via the 
+    // IAuthenticationProvider interface so that GraphServiceClient or AuthHandler can use it.
+    // A significantly enhanced version of this class will in the future be available from
+    // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
+    public class MsalAuthenticationProvider : IAuthenticationProvider
+    {
+        private IConfidentialClientApplication _clientApplication;
+        private string[] _scopes;
+
+        public MsalAuthenticationProvider(IConfidentialClientApplication clientApplication, string[] scopes) {
+            _clientApplication = clientApplication;
+            _scopes = scopes;
+        }
+
+        /// <summary>
+        /// Update HttpRequestMessage with credentials
+        /// </summary>
+        public async Task AuthenticateRequestAsync(HttpRequestMessage request)
+        {
+            var token = await GetTokenAsync();
+            request.Headers.Authorization = new AuthenticationHeaderValue("bearer", token);
+        }
+
+        /// <summary>
+        /// Acquire Token 
+        /// </summary>
+        public async Task<string> GetTokenAsync()
+        {
+            AuthenticationResult authResult = null;
+            authResult = await _clientApplication.AcquireTokenForClient(_scopes)
```

Comment:
Shouln't the the sample be named dotnetcore-console-daemon-sample ? as this is a daemon app only?

Reply:
Yes, the majority of samples in this repo are daemon apps.  Originally one of our blog series contributing authors was going to include an additional 3-4 interactive samples but they ended up being hosted on a different repo.  We could rename this repo but I'll have to look into the impact of refactoring any links, blog posts, and other internal references.

Reply:
GitHub does a good job of maintaining the URL to the previous names of the repos. So links would not be a problem. But I see your point.